### PR TITLE
use AVX instructions for float ops

### PR DESCRIPTION
**Pull request by WalterBright**
For SSE ops, that is. 32 byte vectors not yet supported.

### CODE REVIEW

**Comment by WalterBright** [comment]
Unfortunately, it seems the autotester doesn't have AVX capable computers. So testing of this is rather limited.

**Comment by braddr** [comment]
Quite a few of them do, but definitely not all.  The tests for non-ubiquitous features really need to detect availability of them and rely on statistical coverage.

**Comment by WalterBright** [comment]
I wanted to have `-mavx` on globally to get the best workout for it. I suppose it would be better to put the detection in the test runner and have it simply not throw `-mvax` unless the machine supports it.

https://issues.dlang.org/show_bug.cgi?id=16945

**Comment by WalterBright** [comment]
@braddr any chance of an AVX-512 machine?

**Comment by braddr** [comment]
Unless I'm reading wrong, avx-512 is only currently available in xeon phi x200 co-processors currently.  They won't ship in standard processors until next year.  So, uh, no.

**Comment by WalterBright** [comment]
Guess I'll just have to speculatively support it, then :-)

File: src/backend/cgxmm.c [hunk 1]
```diff
@@ -1344,5 +1348,58 @@ void checkSetVex3(code *c)
...
+ * Determine if operation should be rewritten as a VEX
+ * operation; and do so.
+ * Params:
+ *      c = code
+ *      ty = type of operand
+ */
+
+void checkSetVex(code *c, tym_t ty)
+{
+    if (config.avx || tysize(ty) == 32)
```
Comment (andralex) [inline src/backend/cgxmm.c]: my suicide note will read "I've seen too many `if` statements that spanned entire functions instead of returning early".

**Review by andralex — APPROVED** [review]

**Pull request merged**